src/System.CommandLine/Parsing/CommandResult.cs

```diff
@@ -36,32 +36,147 @@ public sealed class CommandResult : SymbolResult
         /// </summary>
         public IEnumerable<SymbolResult> Children => SymbolResultTree.GetChildren(this);
 
-        internal sealed override int MaximumArgumentCapacity
+        internal override bool UseDefaultValueFor(ArgumentResult argumentResult)
+            => argumentResult.Argument.HasDefaultValue && argumentResult.Tokens.Count == 0;
+
+        /// <param name="completeValidation">Only the inner most command goes through complete validation.</param>
+        internal void Validate(bool completeValidation)
+        {
+            if (completeValidation)
+            {
+                if (Command.Handler is null && Command.HasSubcommands)
+                {
+                    SymbolResultTree.InsertFirstError(
+                        new ParseError(LocalizationResources.RequiredCommandWasNotProvided(), this));
+                }
+            }
+
+            if (completeValidation && Command.HasValidators)
```

Comment:
This looks like it can be moved into the above block since you're rechecking `completeValidation`.